### DISCUSSION TITLE
QueryBuilder: Can I make a conditional paginate using .if?

### DISCUSSION THREAD

**Discussion by AshlinR** [adonisjs/core#4185] [Help]
I want to be able to conditionally return paginated data. At the moment I have the query wrapped in an if() {} block but this results in duplicate code for the condition where there is no pagination, i.e. same code minus the .paginate().

I'd like to be able to write:

   .if(paginate, (query) => {
        query.paginate(page, perPage)
      })

However, this doesn't work. How can I get this to work?

**Comment by teta0412** [comment]
Do you find a way to execute this ?
